src/main/java/com/epam/esm/giftcertificates/utils/GiftCertificatesComplement.java

```diff
@@ -0,0 +1,25 @@
+package com.epam.esm.giftcertificates.utils;
+
+import com.epam.esm.giftcertificates.entity.GiftCertificate;
+import org.apache.commons.lang3.StringUtils;
+
+import java.time.Duration;
+
+
+public class GiftCertificatesComplement {
+    public static GiftCertificate mergingCertificate(GiftCertificate updateGiftCertificate, GiftCertificate certificateFromDB) {
```

Comment:
I can move it to service layer but doesn't that violate the SRP?

Reply:
Because it is logic of merging 2 sertificates, IMO it can be move into model class.
But method parameters should be renamed to not leak implementation details (e.g. fromDB)
I think smth like `targetSertificate` - into which you are going to set fields and `sourceSertificate` - from whick you are going to get fields. 

Moving to service should be fine too IMO, as this is part of data processing logic.
What do you think?

Reply:
Better solution would be to move it to the model class